Plugin.php

```diff
@@ -114,6 +117,23 @@ public function registerFormWidgets()
 
     public function boot()
     {
+        if (PluginManager::instance()->exists('RainLab.Translate')) {
+            Event::listen('backend.form.extendFieldsBefore', function ($widget) {
+                // add localeTitle[lang] fields to MenuItem form
+                if ($widget->model instanceof MenuItem) {
+                    // change type of formwidget for MenuItem form title to mltext
+                    $widget->fields['title']['type'] = 'mltext';
+
+                    foreach (Locale::listAvailable() as $code => $locale) {
+                        $widget->fields["viewBag[localeTitle.$code]"] = [
+                            'cssClass' => 'hidden',
+                            'attributes' => ['data-locale' => $code],
+                        ];
+                    }
+                }
+            });
+        }
```

Comment:
@mjauvin Shouldn't this event callback be done as part of the Translate plugin instead?

Reply:
You're definitely right. I started doing this but then I got stuck in the javascript changes which I couldn't find a way to do from within the RainLab.Translate plugin, so I decided to include it in this PR to show the whole picture. Once we find that it's all good, I can move it back where it really belong.

Reply:
@mjauvin That's cool - I'd probably recommend still storing the changes as a PR to the Translate plugin and linking the PR to this one so that we can test how it all works in its final state.

Reply:
And leave the changes here as well for now or not?